src/overlays/actors/ovl_En_Kakasi2/z_en_kakasi2.c

```diff
@@ -20,21 +42,193 @@ const ActorInit En_Kakasi2_InitVars = {
     (ActorFunc)EnKakasi2_Update,
     NULL,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi2/EnKakasi2_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi2/EnKakasi2_Destroy.s")
+void EnKakasi2_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnKakasi2* this = THIS;
+    s32 pad;
+    f32 temp0;
+    f32 temp1;
+
+    osSyncPrintf("\n\n");
+    // Visit Umeda
+    osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 梅田参号見参！ ☆☆☆☆☆ \n" VT_RST);
+
+    this->switchFlag = this->actor.params & 0x3F;
+    temp0 = (this->actor.params >> 6) & 0xFF;
+    temp1 = this->actor.posRot.rot.z;
+    if (this->switchFlag == 0x3F) {
+        this->switchFlag = -1;
+    }
+    this->actor.unk_1F = 4;
+    this->distance.x = (temp0 * 40.0f) + 40.0f;
+    this->distance.y = (temp1 * 40.0f) + 40.0f;
+
+    // Former? (Argument 0)
+    osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ 元？(引数０) ☆☆☆☆ %f\n" VT_RST, temp0);
+    // Former? (Z angle)
+    osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ 元？(Ｚアングル) ☆☆ %f\n" VT_RST, temp1);
+    // Correction coordinates X
+    osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ 補正座標Ｘ ☆☆☆☆☆ %f\n" VT_RST, this->distance.x);
+    // Correction coordinates Y
+    osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ 補正座標Ｙ ☆☆☆☆☆ %f\n" VT_RST, this->distance.y);
+    // Correction coordinates Z
+    osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ 補正座標Ｚ ☆☆☆☆☆ %f\n" VT_RST, this->distance.z);
+    osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ SAVE       ☆☆☆☆☆ %d\n" VT_RST, this->switchFlag);
+    osSyncPrintf("\n\n");
+
+    this->actor.colChkInfo.mass = 0xFF;
+    this->height = 60.0f;
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actor.flags |= 0x400;
+    this->unk_198 = this->actor.shape.rot.y;
+
+    if (this->switchFlag >= 0 && Flags_GetSwitch(globalCtx, this->switchFlag)) {
+        this->actor.draw = func_80A90948;
+        Collider_InitCylinder(globalCtx, &this->collider);
+        Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+        SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_060065B0, &D_06000214, NULL, NULL, 0);
+        this->actionFunc = func_80A9062C;
+    } else {
+        this->actionFunc = func_80A90264;
+        this->actor.shape.unk_08 = -8000.0f;
+    }
+}
+
+void EnKakasi2_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnKakasi2* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+    //! @bug Skelanime_Free is not called
+}
+
+void func_80A90264(EnKakasi2* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    this->unk_194++;
+
+    if ((BREG(1) != 0) && (this->actor.xzDistFromLink < this->distance.x) &&
+        (fabsf(player->actor.posRot.pos.y - this->actor.posRot.pos.y) < this->distance.y)) {
```

Comment:
maybe `range` would be less generic than `distance`? not sure
(it looks to me like that's the max scarecrow activation range)

Reply:
I like range, what do you think @fig02 ?

Reply:
yeah we were discussing this in the comment above, `maxSpawnRange` is probably good. 
not sure what to name the temps for the range factors passed through params. spawnRangeXZ and spawnRangeY I guess, though its not technically the range itself